src/scene/scene_lua.inl

```diff
@@ -28,73 +28,115 @@ namespace openspace {
+    value
+    );*/
+}
+
+template <class T>
+properties::PropertyOwner* findPropertyOwnerWithMatchingGroupTag(T* prop,
+const std::string tagToMatch)     {
+    properties::PropertyOwner* tagMatchOwner = nullptr;
+    properties::PropertyOwner* owner = prop->owner();
+    
+    if (owner != nullptr) {
+        std::vector<std::string> tags = (std::vector<std::string>)owner->tags();
+        for (std::string currTag : tags) {
+            if (tagToMatch.compare(currTag) == 0) {
+                tagMatchOwner = owner;
+                break;
+            }
+        }
+        
+        //Call recursively until we find an owner with matching tag or the top of the
+        // ownership list
+        if (tagMatchOwner == nullptr)
+            tagMatchOwner = findPropertyOwnerWithMatchingGroupTag(owner, tagToMatch);
+    }
+    return tagMatchOwner;
+}
+
+void applyRegularExpression(lua_State* L, std::regex regex,
+                            std::vector<properties::Property*> properties, int type,
+                            std::string groupName = "")                               {
     using ghoul::lua::errorLocation;
     using ghoul::lua::luaTypeToString;
-
+    bool isGroupMode = (groupName.empty()) ? false : true;
+    
     for (properties::Property* prop : properties) {
         // Check the regular expression for all properties
         std::string id = prop->fullyQualifiedIdentifier();
+
         if (std::regex_match(id, regex)) {
             // If the fully qualified id matches the regular expression, we queue the
             // value change if the types agree
-
+            if (isGroupMode) {
+                properties::PropertyOwner* matchingTaggedOwner = findPropertyOwnerWithMatchingGroupTag(prop,
+                groupName);
+                if (matchingTaggedOwner == nullptr)
+                    continue;
+            }
+            
             if (type != prop->typeLua()) {
                 LERRORC("property_setValue",
                         errorLocation(L) << "Property '" <<
                         prop->fullyQualifiedIdentifier() <<
                         "' does not accept input of type '" << luaTypeToString(type) <<
                         "'. Requested type: '" << luaTypeToString(prop->typeLua()) << "'"
                 );
+            } else {
+               executePropertySet(prop, L);
             }
-            else {
-                prop->setLuaValue(L);
-                //ensure properties are synced over parallel connection
-                std::string value;
-                prop->getStringValue(value);
-/*                OsEng.parallelConnection().scriptMessage(
-                    prop->fullyQualifiedIdentifier(),
-                    value
-                );*/
-            }
-
         }
     }
 }
-}
-
 
-namespace luascriptfunctions {
+bool doesUriContainGroupTag(const std::string command) {
+    std::string name = command.substr(0, command.find_first_of("."));
```

Comment:
I prefer the way I did it since it grabs the first term, and not only finds the curly brackets but ensures that they are at the first and last positions of the string.